rfc/state-file.md

```diff
@@ -0,0 +1,94 @@
+# Constellation state file
+
+During a cluster's lifecycle, Constellation needs to keep track of multiple different values and state information of the cluster,
+like loadbalancer IPs or UID of the cluster.
+
+If this information is configurable by a user, and required to set up the cluster, the user saves it in Constellation's config file.
+Information that is not configurable, and that is generated by the cluster during the creation or init phase is either saved in Terraform output files,
+or in the cluster ID file (`constellation-id.json`).
+
+This creates problems since the relevant information may be split over multiple files, and it is not always clear at what point certain information is available in the cluster ID file.
+Additionally, commands like `upgrade` and `init` currently depend on Terraform files being present in the working directory,
+which may not be the case if the cloud resources were manually created.
```

Comment:
Agreed that it would be nice if we can remove the need for tf state in the workspace with this file. Ideally we would have a command in the CLI to parse TF state into Constellation state, imo.
Sth like `constellation config parse-state <tfstate-path>`.

Reply:
Agreed, but implementation details and integration of parsing of existing terraform state is out of scope for this rfc